srml/support/procedural/src/storage/genesis_config/genesis_config_def.rs

```diff
@@ -114,27 +114,27 @@ impl GenesisConfigDef {
 				.unwrap_or_else(|| quote!( Default::default() ));
 
 			config_field_defs.push(GenesisConfigFieldDef {
-				doc: line.doc_attrs.clone(),
 				name: config_field,
 				typ,
+				attrs: line.doc_attrs.clone(),
 				default,
 			});
 		}
 
 		for line in &def.extra_genesis_config_lines {
-			let doc = line.attrs.iter()
+			let attrs = line.attrs.iter()
 				.filter_map(|a| a.parse_meta().ok())
-				.filter(|m| m.name() == "doc")
+				.filter(|m| m.name() == "doc" || m.name() == "serde")
```

Comment:
You should just store the `line.attrs` directly and don't do any filtering here.

Reply:
So there's two issues. One is that depending on the attribute, you might need to apply it both to the field declaration and the default assignment (for like a `cfg` attribute). Or that some attributes might not make sense at all. The other is that attrs is filtered for doc attrs in the regular storage lines (like not the add_extra_genesis lines) and so it's a `syn::Meta` there, not a `syn::Attribute`. So that would also have to be unrestricted if we made it `Attribute`. What do you think?

Reply:
Okay, then we should do the following:
```
line.attrs.iter().filter_map(|a| a.parse_meta().ok()).try_for_each(|m| if m.name() == "cfg" { Error::new("cfg not supported, bla, good error message") } else { Ok(()) })?;
```

Reply:
If that check is successful, we store the `Vec<Attribute>`, as said before.